packages/Kokkos/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -0,0 +1,338 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
+
+#include <Kokkos_DefaultNode.hpp>
+
+#include <DTK_ConfigDefs.hpp>
+
+#include <DTK_LinearBVH.hpp>
+#include <details/DTK_DetailsTreeTraversal.hpp>
+
+#include <cmath>
+#include <random>
+
+template <typename NO>
+class RandomWithinLambda
+{
+  public:
+    using DeviceType = typename DataTransferKit::BVH<NO>::DeviceType;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    using PointsView = Kokkos::View<double * [3], ExecutionSpace>;
+    using RView = Kokkos::View<double *, ExecutionSpace>;
+
+  public:
+    RandomWithinLambda( PointsView point_coords, RView radii,
+                        DataTransferKit::BVH<NO> bvh,
+                        unsigned int max_n_indices = 10000 )
+        : _point_coords( point_coords )
+        , _radii( radii )
+        , _bvh( bvh )
+        , _max_n_indices( max_n_indices )
+    {
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()( int const i ) const
+    {
+        DataTransferKit::Details::Within within_predicate(
+            {_point_coords( i, 0 ), _point_coords( i, 1 ),
+             _point_coords( i, 2 )},
+            _radii( i ) );
+        int indices[_max_n_indices];
```

Comment:
_max_n_indices is not `static` or `constexpr` so this will not compile (see fetnat buildlog)